changes to cut documentation (#1648)
pace-neutrons/Horace: documentation/user_docs/docs/manual/Cutting_data_of_interest_from_SQW_files_and_objects.rst

`Binning arguments`_ and the presentation of the data.

``proj`` should be a projection type such as ``line_proj``, ``sphere_proj``,

> etc. which contains information about the target coordinate system representation.

Author: not sure about it but fine, may change it.

   to ensure your cut is what you expect.

> * ``[]`` Automatic binning calculations. 

Author: block moved into separate location -- not aware about any complains of too dence. Rather it is too loose. In any case, a "subtitle" would be reasonable compromize.

    Empty brackets indicate that the cut algorithm should identify binning ranges. 

>     The hull which surrounds source image is converted into target coordinate 

Author: region is not surrouning anything. The hull does.

    cut algorithm will try to identify :math:`\phi` range (3-rd coordinate of cylindrical coordinate system)
    of the source cuboid in the cylindrical coordinate system. The number of bins for the :math:`\phi` range
    will be equal to the number of bins in ``q-3`` dimension of the source coordinate system. If 
    ``q-3`` dimension was integrated, the :math:`\phi` dimension of the target cylindrical coordinate system
    will be also integrated.
    

* ``[step]`` Automatic binning calculations with binning step size.
    Single (scalar) number defines a plot axis with bin width equal to the number you specify. 
    The lower and upper limits are calculated by the same algorithm as the binning range in ``[]``-brackets case.  

.. note::

   A value of ``[0]`` is equivalent to ``[n]`` using bin size of the source coordinate system. This may lead
   to strange or incorrect result if target coordinate system is changed significantly as reasonable ``0.01`` 
   q-step may be used as step size for ``-180:180`` :math:`\phi` binning range of spherical or cylindrical

>    coordinate system, creating 36000 bins in :math:`\phi` direction, which is useless and probably fail.

Author: moved

   1]`` will capture pixels from ``-1.5`` to ``1.5`` inclusive.

> * ``[lower, separation, upper, cut_width]``  Multiple cuts.

Author: not in this case -- looks file

    A four-component binning axis defines
    **multiple** cuts with **multiple**  integration limits in the selected direction.
    These components are:

>   * ``lower``      --   minimum cut bin-centre

Author: because it is not distance between bin centers but the coordinate of the first cut

File- and memory-backed cuts
----------------------------

``cut`` generally tries to return its result in memory. 
However, if the resulting object
is sufficiently large with the threshold determined by the
configuration parameters ``mem_chunk_size`` and ``fb_scale_factor`` defined
in :ref:`Horace config <manual/Changing_Horace_settings:Horace Config>` class, the object is 
placed in temporary file and becomes filebacked. This happens if the number of pixels 
in the result exceeds  the value of the product ``mem_chunk_size*fb_scale_factor``. 
See :ref:`manual/Changing_Horace_settings:Changing Horace settings` for more
information about configuring Horace.

Temporary file means that the backing file will be destroyed when the ``sqw`` object,

> which is backed by this file goes out of scope. 

Author: users do care when their file is deleted. And the scope of variables is the second thing a person learning programming learns.

    However, it is advised that besides ``u`` and ``v`` arguments are passed as
    key-value pairs.

>     Alternatively you may define some parameters initially, and define others
>     programmatically:

Reviewer: ```suggestion
    Alternatively you may define some parameters in the constructor, and define others
    later by setting their properties:
```

In order to construct a spherical projection (i.e. a projection in |Q|,
:math:`\theta` (polar angle), :math:`\phi` (azimuthal angle), :math:`E`) we
create a projection in an analogous way to the ``line_proj``, but using the
``sphere_proj`` function:

The complete signature for ``sphere_proj`` is:

.. code-block:: matlab

   proj = sphere_proj([u][, v][, type][, alatt][, angdeg][, offset][, label][, title][, lab1][, lab2][, lab3][, lab4])

where:

- ``u`` The direction of the :math:`z`-axis of the spherical coordinate system. 
    The vector :math:`u` is the reciprocal space vector defining the direction of the :math:`z`-axis 
    of the target spherical coordinate system.  See :ref:`fig_sphere_coodinates` for details.

- ``v``  The direction of the :math:`x`-axis of the spherical coordinate system. 
    The vector :math:`v` is the reciprocal space vector defining the direction of the :math:`x`-axis 
    of the target spherical coordinate system. The reciprocal space vectors :math:`u-v` are not
    always orthogonal so the actual direction of spherical coordinate system :math:`x`-axis lies

>     in the plain defined by :math:`u-v` vectors and is orthogonal to :math:`z`-axis. See 

Author: diagram should use e_x, e_z as I do mention them here and express orther axes trough them.

        Preserve the values of ``u`` and ``v``

  There are 2 possible options for the second and third (angular) components of
  ``type``:

  1. ``'d'``     Degrees

  2. ``'r'``     Radians

>   ..

Author: its commented as we can not at the moment have |Q| in rlu. Will be reverted when we can

.. note::

   Like ``line_proj``, ``sphere_proj`` can be :ref:`defined using
   positional or keyword arguments <poskwarg>`. However the same
   recommendation applies that positionals should only be used to
   define ``u`` and ``v``.

``sphere_proj`` defines a spherical coordinate system, where:

* |Q|

  The radius from the origin (``offset``) in :math:`hkl`

* :math:`\theta`

  The angle measured from :math:`u` to the vector (:math:`\vec{q}`),

>   i.e. :math:`0^{\circ}` is parallel to :math:`u` and :math:`90^{\circ}` is

Reviewer 2: ```suggestion
 so that a \theta value of :math:`0^{\circ}` means that `q` is parallel to :math:`u`; and a \theta value of :math:`90^{\circ}`means that `q` is
```

>   perpendicular to :math:`u`.

Reviewer 2: wondering if \vec should be more liberally sprinkled here

>   The angle measured from :math:`u` to the vector (:math:`\vec{q}`),
>   i.e. :math:`0^{\circ}` is parallel to :math:`u` and :math:`90^{\circ}` is
>   perpendicular to :math:`u`.

Reviewer: ```suggestion
  The angle measured from :math:`\vec{u}` to the vector (:math:`\vec{q}`),
  so a :math:`\theta` value of :math:`0^{\circ}` means that `\vec{q}` is parallel 
  to :math:`\vec{u}`; and a :math:`\theta` value of :math:`90^{\circ}` means that 
  `\vec{q}` is perpendicular to :math:`\vec{u}`.
```

  Mathematically this is defined as:

  .. math::

     \cos\left(\theta{}\right) = \frac{\vec{q}\cdot\vec{u}}{\left|q\right|\cdot\left|u\right|}

* :math:`\phi`

  The angle measured between the :math:`u`-:math:`v` plane to the vector

>   (:math:`\vec{q}`), i.e. :math:`0^{\circ}` lies in the :math:`u`-:math:`v`

Reviewer 2: ```suggestion
  (:math:`\vec{q}`), i.e. :math:`0^{\circ}` means that `q` lies in the :math:`u`-:math:`v`
```

>   plane and :math:`90^{\circ}` is normal to the :math:`u`-:math:`v` plane

Reviewer 2: ```suggestion
  plane and :math:`90^{\circ}` means that `q` is normal to the :math:`u`-:math:`v` plane
```

  .. math::

     \cos\left(\phi{}\right) = \frac{\vec{p} \cdot{} \left(\vec{w} \times{} \vec{u} \right)}
                                    {\vec{q} \cdot{} \vec{w} \times{} \vec{u}}

  where :math:`\vec{p}` is the projection of :math:`\vec{q}` onto the meridian
  plane (the dashed line `below <#fig-sphere-coodinates>`_), given by:

  .. math::

     \vec{p} = \frac{\vec{q}\cdot{}\vec{u}}{\left|u\right|^2}\vec{u}

  ..
     Commented as proof of working not necessary for users

     .. math::

        \cos\left(\phi{}\right) = \frac{\left(\vec{w} \times{} \vec{u} \right) \cdot \vec{p}}
                                       {\left|w\right|\left|u\right|\left|p\right|}

     Where :math:`\left|p\right| = \left|q\right|`

     .. math::

        \cos\left(\phi{}\right) = \frac{\left(\vec{w} \times{} \vec{u} \right) \cdot \vec{p}}
                                       {\left|w\right|\left|u\right|\left|q\right|}

* :math:`E`

  The energy transfer as defined in ``line_proj``

.. note::

   - :math:`\theta` lies in the range between :math:`0^{\circ}` and
     :math:`180^{\circ}`.
   - :math:`\phi` lies in the range between :math:`-180^{\circ}` and
     :math:`180^{\circ}`.

>    in radians, that is:

Reviewer 2: ```suggestion
   In radians, that is:
```

   - :math:`\theta` lies in the range between :math:`0` and :math:`\pi`
   - :math:`\phi` lies in the range between :math:`-\pi` and :math:`\pi`.

.. _fig_sphere_coodinates:

.. figure:: ../images/sphere_proj_coordinates.png
   :align: center
   :width: 400px
   :alt: spherical coordinate system.

   Spherical coordinate system used by ``sphere_proj``

..
   TODO: Move into relevant developer documentation section

>    In practice, Horace uses Matlab methods ``cart2sph`` and ``sph2cart`` to convert array of vectors expressed

Reviewer 2: ```suggestion
   In practice, Horace uses Matlab methods ``cart2sph`` and ``sph2cart`` to convert an array of vectors expressed
```

.. math::

   \cos\left(\theta{}\right) = \frac{\vec{Q}\cdot\vec{e_z}}{\left|Q\right|\cdot\left|e_z\right|}

* :math:`\phi` --  is the angle measured between the :math:`e_x`-:math:`e_z` plane to the vector
  (:math:`\vec{q}`), i.e. :math:`0^{\circ}` lies in the :math:`e_x`-:math:`e_z`
  plane and :math:`90^{\circ}` is normal to :math:`e_x`-:math:`e_z` plane
  (i.e. parallel to :math:`e_y`). Mathematically this is defined as:

.. math::

   \sin\left(\phi{}\right) = -\frac{[\vec{Q}\times\vec{e_z}]\cdot\vec{e_x}}{\left|[\vec{Q}\times\vec{e_z}]\right|}

* :math:`E`   is the energy transfer as defined in ``line_proj``

.. note::

   :math:`\theta`  lies in the range between :math:`0^{\circ}` and :math:`180^{\circ}` and :math:`\phi` 
   is in the range between :math:`-180^{\circ}` and :math:`180^{\circ}`. The ``sphere_proj`` settings
   allow to change these values to radians so it :math:`0 \leq \theta \leq \pi` and :math:`-\pi \leq \phi \leq \pi`.

.. _fig_sphere_coodinates:

.. figure:: ../images/sphere_proj_coordinates.jpg
   :align: center
   :width: 400px
   :alt: spherical coordinate system.

   Spherical coordinate system used by ``sphere_proj``

> In practice, Horace uses Matlab methods ``cart2sph`` and ``sph2cart`` to convert array of vectors expressed 

Author: they arrarently do as they asked for that.

   The form of the arguments to ``cut`` is still the same (see: `Binning
   arguments`_). However:

   - |Q| runs from :math:`[0, \infty)` -- attempting to use a |Q| with a minimum
     bound less than :math:`0` will fail.
   - :math:`\theta` runs between :math:`[0, 180]` -- requesting binning outsize these ranges will fail.

>    - :math:`\phi` runs between :math:`[-180, 180]` -- requesting binning outsize these ranges will fail.

Author: 1) 0 is included

   centre at :math:`[0,-1,1]`.

.. _cylindrical_projection:

> Cylindrical Projections

Reviewer: While I am not repeating them here, all the comments in the spherical section apply here and should be addressed.

^^^^^^^^^^^^^^^^^^^^^^^^

In order to construct a cylindrical projection (i.e. a projection in :math:`Q_{tr}`, :math:`Q_{||}`,
:math:`\phi` (azimuthal angle) and :math:`E`) coordinate system we create a projection in a way
similar to the ``line_proj``, but using the ``cylinder_proj`` class.

The complete signature for ``cylinder_proj`` is similar to ``sphere_proj``:

.. code-block:: matlab

   proj = cylinder_proj([u][, v][, type][, alatt][, angdeg][, offset][, label][, title][, lab1][, lab2][, lab3][, lab4])

where:

- ``u`` The direction of the :math:`z`-axis of the cylindrical coordinate system. 

>     The vector :math:`u` is the reciprocal space vector defining the direction of the :math:`z`-axis 

Reviewer: Might be better to define it as the principal axis of the cylinder around which the $\theta$ is specified

    of the target cylindrical coordinate system. See :ref:`fig_cylinder_coodinates` for details.

> - ``v``  The direction of the :math:`x`-axis of the cylindrical coordinate system. 

Reviewer: Again, it might be better to define this as a radial direction forming the basis for $\theta$

Author: do not know, looks ok to me.

where:

- ``u``

  The vector :math:`\vec{u}` is the reciprocal space vector defining the
  polar-axis of the cylindrical coordinate system along which
  :math:`\vec{q}_{\|}` is measured.

  See the :ref:`diagram below <fig_cylinder_coodinates>` for details.

- ``v``

  The vector :math:`\vec{v}` is the reciprocal space vector which defines the second
  component of the :math:`u`-:math:`v` plane from which :math:`\phi` is measured.

  See the :ref:`diagram below <fig_cylinder_coodinates>` for details.

.. note::

  The reciprocal space vectors :math:`u`-:math:`v` are not necessarily
  orthogonal so the actual axis from which :math:`\phi` is measured lies in the
  plane defined by :math:`u`-:math:`v` vectors, orthogonal to :math:`u`.

.. note::

   By default a ``cylinder_proj`` will define its principal axes :math:`u` and
   :math:`v` along the :math:`hkl` directions :math:`[1,0,0]` and
   :math:`[0,1,0]` respectively.

- ``type``

  Three character string denoting the the projection normalization of each
  dimension, one character for each directions, e.g. ``'aad'`` or ``'aar'``.

>   At the moment there is only one possible option is implemented for the length

Reviewer 2: ```suggestion
  At the moment there is only one possible option implemented for the length
```

.. note::

   You cannot change the binning in a ``dnd`` object, i.e. you can only set the
   integration ranges, and have to use ``[]`` for the plot axis. The only option
   you have is to change the range of the plot axis by specifying
   ``[lo1,0,hi1]`` instead of ``[]`` (the '0' means 'use existing bin size').

Changing Projections
^^^^^^^^^^^^^^^^^^^^

When a ``cut`` will change projections (i.e. the source projection is different

> to the target projection or the ) there are a few things to be aware of

Reviewer 2: incomplete sentence

particularly when you specify automatic (``[]``, ``[step]``) binning arguments.

.. rubric:: Binning range meaning

When you specify the binning ranges these are defined in the the "target"

> (desired/provided) coordinate system. E.g. in cutting from a linear to a

Reviewer 2: consider alternative to provided

> (desired/provided) coordinate system. E.g. in cutting from a linear to a

Reviewer: ```suggestion
(desired) coordinate system. E.g. in cutting from a linear to a
```
